### PR TITLE
[JENKINS-57078] pipeline integration

### PR DESCRIPTION
**Pull request by vtitov**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@vtitov sorry, I do not longer maintain these repositories.
I might be able to land this change (esp. if @TonyNoble is interested), but it has to be resubmitted to https://github.com/jenkinsci/envinject-api-plugin . This repository exists for compatibility reasons only

File: src/main/java/org/jenkinsci/lib/envinject/service/EnvInjectActionPipelinedRetriever.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author Gregory Boissinot
+ * @deprecated The actual version of this API class is located in EnvInject API Plugin
+ */
+@Deprecated
+//@Restricted(NoExternalUse.class)  // TODO re-enable deprecation & restriction
+public class EnvInjectActionPipelinedRetriever {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/lib/envinject/service/EnvInjectActionPipelinedRetriever.java]: This class already has a Pipeline-compatible implementation here: https://github.com/jenkinsci/envinject-api-plugin/blob/master/src/main/java/org/jenkinsci/plugins/envinjectapi/util/EnvInjectActionRetriever.java

File: src/main/java/org/jenkinsci/lib/envinject/service/EnvVarsPipelinedResolver.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+import java.util.Map;
+
+/**
+ * @author Gregory Boissinot
+ * @deprecated The actual version of this API class is located in EnvInject API Plugin
+ */
+@Deprecated
+@Restricted(NoExternalUse.class)
+@SuppressFBWarnings(value = "SE_NO_SERIALVERSIONID", justification = "Deprecated code")
+public class EnvVarsPipelinedResolver implements Serializable {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/lib/envinject/service/EnvVarsPipelinedResolver.java]: https://github.com/jenkinsci/envinject-api-plugin/blob/master/src/main/java/org/jenkinsci/plugins/envinjectapi/util/EnvVarsResolver.java was refactored to support  Pipeline. Looks like there is no practical differences between classes

**Comment by TonyNoble** [comment]
Given @oleg-nenashev 's comments - do we need to worry about applying these changes at all, or simply consider envinject-lib mothballed and encourage the use of envinject-api instead?

I might be willing to take this on if required in the near future if we feel there's still a need for development, but need the dust to settle on current work first.

**Comment by oleg-nenashev** [comment]
@TonyNoble all plugins should be using EnvInject API. That plugin still includes this library due to some compatibility reasons (mostly-XTrigger Lib), but I am inclined doing a breaking change once all dependent code is migrated

**Comment by oleg-nenashev** [comment]
Closing this pull request. Unless I am missing something, it is not needed
@vtitov please reopen that if you see any cases not addressed by EnvInject API

**Pull request closed without merging**